client/components/payment-activity/index.tsx

```diff
@@ -68,9 +55,16 @@ const PaymentActivity: React.FC = () => {
 
 	const { selectedCurrency } = useSelectedCurrency();
 
+	const [ dateRange, setDateRange ] = useState( {
```

Comment:
Curious what state is essential here in outer component (payment widget) and what we can hide or manage in `DateRangePicker`. 

Feels like the outer component and data fetcher need the dates, and the preset might be mostly associated with the date selector (as a way to generate dates). 

What's the minimum we need out here? 

Do we need a separate module to decouple the date calculations from the date picker component, so we can use for defaults (i.e. `last_7_days` ⇒ real dates), and make the date picker 100% "dumb"?

Reply:
As we are discussing in Slack, it may be suitable to simplify the date-range-preset logic and state into a custom hook.

Then the parent `PaymentActivity` component can consume this hook and render `InlineLabelSelect` with the date-preset selection abstracted away.

I prepared an example of what this might look like in #8939.

Reply:
The changes proposed by @Jinksi  were very helpful! I have added them to the working branch here - [c373c4f](https://github.com/Automattic/woocommerce-payments/pull/8927/commits/c373c4f86c250fd3e65d3d522f94b64297be4192)

Reply:
Nice! Might be good to add a docblock to that hook to capture the intention.

Reply:
👍 I've added a docblock in ddee0f7